### PR TITLE
NanoUI synth airlock control, ERT equipment rework, fixes

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Converts the synthetic airlock control menu to NanoUI (from Bay, by PsiOmegaDelta: https://github.com/Baystation12/Baystation12/pull/8340).
2) Refactors the power/electify procs on doors.
3) Updates the ion storm event (https://github.com/Baystation12/Baystation12/pull/8345).
4) Fixes https://github.com/ParadiseSS13/Paradise/issues/495.
6) Fixes a portable turret runtime.
7) AI's can now properly examine meters.
8) Continues Dave's ERT changes by reworking the way they're called. Players and admins can now pick which type of ERT they'd like to send, and the members can only pick from the roles that have been specified. Also adds the Janitorial ERT (with purple hardsuits) for clean-up duties. I've also reshuffled some of the equipment the ERT spawns with so it's even less effort for them to start (they start with the helmets on, with the oxygen tank in their pocket), and I've also made some modifications to their overall equipment. All that remains now is ensuring the CentCom armories make a bit more sense.
9) Adds the ore satchel of holding (by Xhuis, https://github.com/tgstation/-tg-station/pull/8122).
10) Adds Night Vision Meson Goggles (by GunHog, https://github.com/tgstation/-tg-station/pull/7705).
11) Possible fix for https://github.com/ParadiseSS13/Paradise/issues/501.

### CODE REVIEW

**Comment by PsiOmegaDelta** [comment]
Thought I should mention that I've updated my PR just a little bit, just so you're aware.

**Comment by Markolie** [comment]
Thanks for the heads up!

**Comment by Fox-McCloud** [comment]
@Markolie 

Not saying to utilize TG just because of it being TG, but TG's Ion storm is handled by a proc "generate_ion_law()"; the amount of law diversity and complexity is much much higher and it allows us to implement the Toy AI and Toy AI traitor module.

Also huge huge thumbs down on utilizing Bay's door system; it's slow and still uses the outdated emag system of emag = unuseable door forever.

**Comment by Markolie** [comment]
Not going to port over /tg/'s ion storm event, as that's not within the scope of this fix.

And this PR doesn't change our door system, it only consolidates some procs. Emagging wasn't changed, and that's not within the scope of this PR either.

**Comment by Fox-McCloud** [comment]
I know it doesn't, but I'm saying that your fix is going to mess with door emagging and the changes you've made to doors isn't consolidation of procs (although that is accomplished to a degree) so much as it's out and out implementing the way Bay handles doors--which means waiting forever for them to open and close, far more powerful doorcrushing by the AI.

**Comment by Markolie** [comment]
The door opening speed/crushing was changed back in the last commit (as that is something I overlooked, so that's a fair point). The way Bay handles door was refactored by PsiOmegaDelta in the PR I linked, and it is vastly superior to the way we used to handle our doors and the way /tg/ currently handles our door. This is solely a backend change: it is not noticeable on the front-end. If we want to port over /tg/ emagging, this doesn't stand in the way of that at all.

And again, emagging was not changed. It works identically to the way it used to.

**Comment by Fox-McCloud** [comment]
@Markolie 

Very well, the primary concern is that the emagged rework can be factored in and that TG+Goon's quick open system remains intact; no sense in keeping players waiting 2 seconds at each door.

**Comment by Markolie** [comment]
Yea, definitely. The speed's a good thing.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -1063,34 +872,26 @@ About the new airlock wires panel:
...
 	else if(istype(src, /obj/machinery/door/airlock/clown))
 		playsound(src.loc, 'sound/items/bikehorn.ogg', 30, 1)
 	else if(istype(src, /obj/machinery/door/airlock/mime))
+		// Play nothing
 	else
 		playsound(src.loc, 'sound/machines/airlock.ogg', 30, 1)
 	if(src.closeOther != null && istype(src.closeOther, /obj/machinery/door/airlock/) && !src.closeOther.density)
 		src.closeOther.close()
-	// This worries me - N3X
-	if(autoclose  && normalspeed)
```
Comment (Fox-McCloud) [inline code/game/machinery/doors/airlock.dm]: Doesn't this bit of code (from 1078 to 1084) still need to be in somewhere to retain the behavior of airlocks closing very quickly, after opening (overridden safeties)?

Comment (Markolie) [inline code/game/machinery/doors/airlock.dm]: When I tested it it let me through quickly, but I'll double-check.

Comment (Fox-McCloud) [inline code/game/machinery/doors/airlock.dm]: Oh! I meant a different mechanic.

When you override the doors' safeties and open it, the door will almost immediately shut again (this has nothing to do with the "walk through door halfway through animation thing); was wondering if that bit of code was necessary to retain that behavior.

Comment (Markolie) [inline code/game/machinery/doors/airlock.dm]: It is not, no. That's handled elsewhere now (just tested it).

File: code/game/response_team.dm [hunk 2]
```diff
@@ -345,23 +385,78 @@ proc/trigger_armed_response_team(var/force = 0)
...
+					M.equip_to_slot_or_del(new /obj/item/weapon/rcd/combat(M), slot_in_backpack)
+					M.equip_to_slot_or_del(new /obj/item/weapon/rcd_ammo/large(M), slot_in_backpack)
+					M.equip_to_slot_or_del(new /obj/item/weapon/rcd_ammo/large(M), slot_in_backpack)
+					M.equip_to_slot_or_del(new /obj/item/stack/sheet/metal/full(M), slot_in_backpack)
+					M.equip_to_slot_or_del(new /obj/item/stack/sheet/glass/full(M), slot_in_backpack)
+					M.equip_to_slot_or_del(new /obj/item/stack/sheet/plasteel/full(M), slot_in_backpack)	
+					
+					M.equip_to_slot_or_del(new /obj/item/weapon/storage/briefcase/inflatable(M), slot_r_hand)
+					
+				if("Janitorial")
```
Comment (Fox-McCloud) [inline code/game/response_team.dm]: Oh gosh.

**Comment by Fox-McCloud** [comment]
Wonder if display_contents_with_number can be set to 1 for bags yet?

File: code/game/response_team.dm (hunk 2 above)
Comment (Markolie) [inline code/game/response_team.dm]: You think it's too much :P? The old ERT did have access to six stacks of each, and generally an RCD alone isn't enough.

**Comment by Markolie** [comment]
I have no idea if that needs any refactoring.

**Comment by Fox-McCloud** [comment]
It's being actively used on TG without issue; storage.dm is more or less the same there as it is here (with a few differences).

it's not super important that it be enabled, but it is pretty doggone convenient when you have a ton of different ores and you want an idea of how much space you have (or don't).

File: code/game/response_team.dm (hunk 2 above)
Comment (Fox-McCloud) [inline code/game/response_team.dm]: nah, not that. My "oh gosh" was to an ERT Janitor XP

**Comment by Spacemanspark** [comment]
Are you referring to how /tg/ shows ores in a Minecraft-like way (Stacking), Fox?
That would be very nice to have.

File: code/game/response_team.dm (hunk 2 above)
Comment (Markolie) [inline code/game/response_team.dm]: Ahh :P. Well, we'll see if players abuse it for fun. If so, we can just limit to to admin-only.

**Comment by Fox-McCloud** [comment]
@Spacemanspark 

Yes, exactly that.

**Pull request merged**